File: src/java/Node.java
public class Node implements Comparable<Node> {
    char character;
    int frequency;
    Node left;
    Node right;
    
    Node(char character, int frequency) {
        this.character = character;
        this.frequency = frequency;
        this.left = null;
        this.right = null;
    }

    public boolean isLeaf() {
        if (this.character == '$') {
            return false;
        } else {
            return true;
        }
    }

    public int compareTo(Node other) {
        int frequencyCompare = Integer.compare(this.frequency, other.frequency);

        if (frequencyCompare == 0) {
            return 1;
        }

        return frequencyCompare;
    }

    public String toString() {
        StringBuilder str = new StringBuilder();

        str.append(this.character);
        
        if (this.left != null) {
            str.append(this.left.toString());
        }

        if (this.right != null) {
            str.append(this.right.toString());
        }

        return str.toString();
    }
}

File: python/Node.py
class Node:
    def __init__(self, character, frequency):
        self.character = character
        self.frequency = frequency
        self.left = None
        self.right = None

    def __eq__(self, other):
        return self.frequency == other.frequency
    
    def __lt__(self, other):
        return self.frequency < other.frequency
    
    def __repr__(self, level=0):
        indent = "    " * level
        f = f"{indent}(\"{self.character}\" | {self.frequency})\n"
        if self.left:
            f += f"{self.left.__repr__(level+1)}"
            f += f"{self.right.__repr__(level+1)}"  
        return f